### ISSUE TITLE
Sandbox.reset() no longer invokes resetHistory() on contained spies

### ISSUE THREAD

**Issue by dpmott** [sinonjs/sinon#1770]
* Sinon version : 5.0.1
* Environment   : Windows / Angular 4.4.6
* Example URL   :
* Other libraries you are using:

**What did you expect to happen?**
I expect calls to `sandbox.reset()` to invoke `spy.resetHistory()` for spies in the collection.

**What actually happens**
Calls to `sandbox.reset()` invoke `reset()` on all fakes in its collection.  Now that `reset()` has been removed from the spy API and no longer aliases `resetHistory()`, the spy history is no longer reset.

**How to reproduce**
```
    // in beforeAll:
    navigateSpy = sandbox.spy(TestBed.get(Router), 'navigate');
    // in afterEach:
    sandbox.reset();
    // in tests, cause a call to Router.navigate and then check spy call count
```

Related issue:
Issue https://github.com/sinonjs/sinon/issues/1712 is technically resolved with release 5.0.1, but it breaks the ability to reset the call count for spies contained within a sandbox when `sandbox.reset()` is called.

Proposed fix:
In `lib\sinon\sandbox.js`:
Change implementation of `sandbox.reset` to be:
```
    sandbox.reset = function reset() {
        this.resetHistory();
        this.resetBehavior();
    };
```

This matches the implementation in `lib\sinon\stub.js`

**Comment by mroderick** [comment]
Thank you for your contribution to Sinon.JS. I have created a PR #1772 to address the issue.

**Comment by mroderick** [comment]
The fix has been published to npm as `sinon@5.0.2`
